extractor/filesystem/osv/metadata.go

```diff
@@ -21,3 +21,25 @@ type Metadata struct {
 	Ecosystem string
 	CompareAs string
 }
+
+type DepGroups interface {
```

Comment:
What do we do if extractors that store dep groups need to store other metadata too? e.g. the SCALIBR requirements.txt extractor also stores HashCheckingModeValues and thus has its own metadata type:

https://github.com/google/osv-scalibr/blob/main/extractor/filesystem/language/python/requirements/metadata.go

Would we just add the DepGroups to the existing Metadata type?

Reply:
Yep that's the idea, you can just implement DepGroups interface to the existing type.